lib/dash/segment_template.js

```diff
@@ -426,7 +453,7 @@ shaka.dash.SegmentTemplate = class {
           references.push(reference);
           nextPosition++;
         }
-        if (presentationTimeline.getSegmentAvailabilityEnd() >= periodEnd &&
+        if (availabilityStartTime > getPeriodEnd() &&
```

Comment:
This change is needed to make the timer continue firing until the entire period is prior to the segment availability window so that all references will be evicted. Prior to this the timer stopped firing prematurely causing a large number of these error messages 

"Assertion failed: Should not see evicted segments after available segments"

Reply:
Ah, I see!  That makes sense.  Good catch, and thank you for the fix!

Can you please put your PR comment into a code comment, so that some well-meaning person doesn't some day revert your change to "optimize" something?  Something like this would work for me:

```js
// The timer must continue firing until the entire period is
// unavailable, so that all references will be evicted.
```